### PR TITLE
Shortcircuit packets to self in exec-op & Resend pending requests for validation

### PR DESCRIPTION
**Pull request by davidrusu**

### CODE REVIEW

File: src/deterministic_brb.rs [hunk 1]
```diff
@@ -245,6 +245,33 @@ impl<A: Actor<S>, SA: SigningActor<A, S>, S: Sig, BRBDT: BRBDataType<A>>
         Ok(packets)
     }
 
+    /// Resend any RequestValidation packets that have not yet received enough signatures.
+    #[allow(clippy::type_complexity)]
+    pub fn resend_pending_validation_requests(
+        &self,
+    ) -> Result<Vec<Packet<A, S, BRBDT::Op>>, Error<A, S, BRBDT::ValidationError>> {
+        let mut packets = Vec::new();
+        for (msg, sigs) in self.pending_proof.iter() {
```
Comment (dan-da) [inline src/deterministic_brb.rs:254]: pending_proof is not inserted into until SignedValidated is received.  So I believe this still leaves a window between the broadcast in exec_op() until we get SignedValidated.

I am thinking that we need to somehow mark our actor/self with "source-op-in-flight" inside exec_op, and remove that after supermajority-of-supermajority.

could be I'm missing something though... ;-)

File: src/deterministic_brb.rs [hunk 2]
```diff
@@ -280,7 +307,20 @@ impl<A: Actor<S>, SA: SigningActor<A, S>, S: Sig, BRBDT: BRBDataType<A>>
         };
 
         info!("[BRB] {} initiating bft for msg {:?}", self.actor(), msg);
-        self.broadcast(&Payload::BRB(Op::RequestValidation { msg }), self.peers()?)
+        let (mut self_packets, mut others_packets): (Vec<_>, Vec<_>) = self
```
Comment (dan-da) [inline src/deterministic_brb.rs:310]: I think  we need a comment here or at top of func about short-circuit logic / purpose.

File: src/deterministic_brb.rs (hunk 1 above)
Comment (davidrusu) [reply src/deterministic_brb.rs:254]: > pending_proof is not inserted into until SignedValidated is received. So I believe this still leaves a window between the broadcast in exec_op() until we get SignedValidated.

With the short-circuiting in exec_op, we will process the RequestValidation and SignedValidated from ourselves before we return.

Comment (dan-da) [reply src/deterministic_brb.rs:254]: I see.   so pending_proof is inserted into before exec_op() returns.  ok, that works afaict.  nice.

yeah, that'd be a good thing to comment in exec_op also, cuz not so obvious.

**Review by dan-da — APPROVED** [review]

**Pull request merged**